packages/workspace/src/browser/workspace-service.ts

```diff
@@ -71,13 +71,13 @@ export class WorkspaceService implements FrontendApplicationContribution {
 
         this.watcher.onFilesChanged(event => {
             if (this._workspace && FileChangeEvent.isAffected(event, new URI(this._workspace.uri))) {
-                this.updateWorkspace();
+                this.updateWorkspaceRoots();
```

Comment:
changes in `WorkspaceService` does not seem to do anything with fixing the issue and can be just reverted

Reply:
The original problem was caused because of a misunderstanding of the meaning of onWorkspaceChange.  Although I added a unit test to test changes inside Git repositories, this would not prevent regressions caused by developers misunderstanding the intent of the workspace change event.  For that reason I renamed things to clarify intent.  However this is not important and we do need the fix so I have reverted the internal changes to WorkspaceService.